### ISSUE TITLE
Argument checks

### ISSUE THREAD

**Issue by nevrome** [sidora-tools/sidora.cli#25]
R is a dynamically typed language and there are no default input checks for functions. Stephans interface criticism some weeks ago was a wake-up call and made me aware how wonky R code often feels just because of that: Functions sometimes fail and produce the weirdest output if used with the wrong input. Of course this design choice also has big advantages and allows for really crazy dynamic code, but for many interfaces it's harmful.

I just took a look at the multitude of packages available to compensate for this and came to the conclusion, that [checkmate](https://github.com/mllg/checkmate) might be worth a try.

**Comment by jfy133** [comment]
Looks good!

**Comment by stschiff** [comment]
Yes, certainly needed in a language like R, although I haven't looked into the exact interface.
